### PR TITLE
Added Line Slice and Nearest Point on Line functionalities

### PR DESCRIPTION
**Pull request by kapitan-iglu**
Finally I implemented lineSlice() functionality and nearestPointOnLine() (as a dependency) based on turfjs idea to fulfill requirements in my project (#82). 

What is missing:
1. Examples - I'm not familiar with documentation platform (sphinx, jupyter, ...), so if you like, add example(s)
1. I'm quite new in turf* and geojson world, so if you see possible optimizations in my code, please modify it so I can learn something new
1. Tests - same as above - never done any pytest but I'll be glad to see how it should be done

### CODE REVIEW

**Comment by omanges** [comment]
Please update your PR with latest changes

**Comment by omanges** [comment]
@kapitan-iglu  try to add a test for the new functionality in test_misc.py, just add a new test function in the test_misc.py and that test will br automatically executed by pytest, you can refer other test functions in it, it is quite simple 😉

**Comment by kapitan-iglu** [comment]
Synced with upstream repo and added tests as requested. After fixing #86 `test_nearest_point_on_line()` can be adapted to use MultiLineString.

File: turfpy/misc.py [hunk 1]
```diff
@@ -238,3 +240,139 @@ def bbox(coords1, coords2):
...
+    :param line: line to slice
+    :return: sliced line as LineString Feature
+    """
+
+    if not line or get_type(line) != "LineString":
+        raise Exception("line must be a LineString")
+
+    coords = get_coords(line)
+    startVertex = nearest_point_on_line(line, startPt)
+    stopVertex = nearest_point_on_line(line, stopPt)
```
Comment (sackh) [inline turfpy/misc.py:365]: please do not use camel case for variable names.

File: turfpy/misc.py [hunk 2]
```diff
@@ -238,3 +240,139 @@ def bbox(coords1, coords2):
...
+
+    if startVertex["properties"]["index"] <= stopVertex["properties"]["index"]:
+        ends = [startVertex, stopVertex]
+    else:
+        ends = [stopVertex, startVertex]
+
+    clipCoords = [get_coord(ends[0])]
```
Comment (sackh) [inline turfpy/misc.py:372]: please do not use camel case for variable names.

File: turfpy/misc.py [hunk 3]
```diff
@@ -238,3 +240,139 @@ def bbox(coords1, coords2):
...
+    :param stopPt: stopping point
+    :param line: line to slice
+    :return: sliced line as LineString Feature
+    """
+
+    if not line or get_type(line) != "LineString":
+        raise Exception("line must be a LineString")
+
+    coords = get_coords(line)
+    startVertex = nearest_point_on_line(line, startPt)
```
[the remark this answers is not in the document]
Comment (kapitan-iglu) [reply turfpy/misc.py]: Too much inspiration in turfjs... I was expecting that linter checks also for this. Nevermind, it's corrected.

**Review by sackh — APPROVED** [review]

**Pull request merged**